crates/factor-outbound-mysql/src/client.rs

```diff
@@ -0,0 +1,308 @@
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
+use mysql_async::consts::ColumnType;
+use mysql_async::prelude::{FromValue, Queryable as _};
+use mysql_async::{from_value_opt, Conn as MysqlClient, Opts, OptsBuilder, SslOpts};
+use spin_core::async_trait;
+use spin_world::v2::mysql::{self as v2};
+use spin_world::v2::rdbms_types::{
+    self as v2_types, Column, DbDataType, DbValue, ParameterValue, RowSet,
+};
+use url::Url;
+
+#[async_trait]
+pub trait Client {
```

Comment:
For these sort of single-consumer traits you can save yourself some type noise by adding "supertrait" constraints:

```suggestion
pub trait Client: Send + Sync + 'static {
```